packages/lib/models/Tag.test.js

```diff
@@ -156,4 +156,38 @@ describe('models/Tag', function() {
 		expect(commonTagIds.includes(tagc.id)).toBe(true);
 	}));
 
+	it('should sort tags for setTags, Sidebar and TagList', (async () => {
+		const folder1 = await Folder.save({ title: 'folder1' });
+		const note1 = await Note.save({ title: 'ma note 0', parent_id: folder1.id });
+
+		// original/sorted order
+		const tag1 = await Tag.save({ title: '#coding' });
+		const tag2 = await Tag.save({ title: '#house' });
+		const tag3 = await Tag.save({ title: '#wait' });
+		const tag4 = await Tag.save({ title: '@⏲15 min' });
+		const tag5 = await Tag.save({ title: '@⏲30 min' });
+		const tag6 = await Tag.save({ title: '@⏲60 min' });
+
+		await Tag.addNote(tag4.id, note1.id);
+		await Tag.addNote(tag2.id, note1.id);
+		await Tag.addNote(tag1.id, note1.id);
+		await Tag.addNote(tag6.id, note1.id);
+		await Tag.addNote(tag3.id, note1.id);
+		await Tag.addNote(tag5.id, note1.id);
+
+		// This test checks the sorting functionality for tags in all three files(setTags, Sidebar and TagList)
+		// commonTags contains unsortedTags
+		const commonTags = await Tag.commonTagsByNoteIds([note1.id]);
+
+		// defaultTags contains only the id and title of all the unsorted tags
+		const defaultTags = commonTags.map(a => {
+			return { value: a.id, label: a.title };
+		});
+
+		// sortedTagsTitles contains the titles of all the sorted tags
+		const sortedTagsTitles = Tag.sortTags(defaultTags).map(a => a.label);
+
+		expect(sortedTagsTitles).toEqual(['@⏲15 min', '@⏲30 min', '@⏲60 min', '#coding', '#house', '#wait']);
+		expect(sortedTagsTitles).not.toEqual(['@⏲15 min', '#house','#coding', '@⏲60 min', '#wait', '@⏲30 min']);
+	}));
```

Comment:
Please add more tests - empty list, tags with no title property, etc. Read the code you've copied and pasted in sortTags and make sure you test all cases.